File: Programmers/[81301] 숫자 문자열과 영단어/JFe/81301.py
def solution(s):
    answer, word = 0, ''
    number = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    # 문자열 s에서 글자 하나씩 분석
    for c in s:
        # 숫자인 경우
        if c.isdigit():
            answer *= 10
            answer += int(c)
        # 숫자가 아닌 경우
        else:
            word += c
        # word가 완성된 경우
        if word in number:
            answer *= 10
            answer += number.index(word)
            word = ''
    return answer

File: Programmers/[81301] 숫자 문자열과 영단어/KingDonggyu/81301.js
function solution(s) {
  const lexicon = new Map();

  lexicon
    .set('zero', 0)
    .set('one', 1)
    .set('two', 2)
    .set('three', 3)
    .set('four', 4)
    .set('five', 5)
    .set('six', 6)
    .set('seven', 7)
    .set('eight', 8)
    .set('nine', 9);

  let answer = '';
  let string = '';

  for (const ch of s) {
    if (+ch || ch === '0') {
      answer += ch;
      continue;
    }

    string += ch;

    if (lexicon.has(string)) {
      answer += lexicon.get(string);
      string = '';
    }
  }

  return +answer;
}
